sync/src/main/java/org/bf2/sync/ManagedKafkaSync.java

```diff
@@ -103,6 +103,25 @@ public void syncKafkaClusters() {
             String localKey = Cache.namespaceKeyFunc(remoteManagedKafka.getMetadata().getNamespace(), remoteManagedKafka.getMetadata().getName());
             ManagedKafka existing = lookup.getLocalManagedKafka(localKey);
 
+            // special handling if the control plane indicates paused
+            if (remoteManagedKafka.getAnnotation(ManagedKafkaKeys.Annotations.PAUSE_RECONCILIATION).map(Boolean::valueOf).orElse(false)) {
+                if (!remoteManagedKafka.getSpec().isDeleted()) {
+                    log.debugf("Taking no action on ManagedKafka %s as the control plane has paused it", Cache.metaNamespaceKeyFunc(remoteManagedKafka));
+                    continue;
+                }
+                if (existing != null) {
+                    if (!existing.getSpec().isDeleted()) {
+                        log.debugf("Passing just the delete flag to the ManagedKafka %s", Cache.metaNamespaceKeyFunc(remoteManagedKafka));
+                        kubeClient.resources(ManagedKafka.class)
+                                .inNamespace(existing.getMetadata().getNamespace())
+                                .withName(existing.getMetadata().getName())
+                                .patch(PatchContext.of(PatchType.JSON),
+                                        "[{\"op\": \"replace\", \"path\":\"/spec/deleted\", \"value\":true}]");
```

Comment:
little confused about the logic here, if the control plane pause is true, and if the local instance exists and is not deleted it sets "spec/delete" to true?

Reply:
You'll only do that if the remote is marked as deleted:

```
if remote paused {
  if remote not deleted {
    move to the next item to skip main reconciliation
  }
  if local exists {
    if local is not deleted {
      mark local deleted
    }
    move to the next item to skip main reconciliation
  }
} 
```

Reply:
Thanks! remote deleted and paused at the same time seems like a weird case imo.

Reply:
`deleted` is a status in the database. 
`pause-reconciliation` directly ties to whether some resources e.g TLS certificates where able to be fetched or not. These resources are fetched each time a ManagedKafka is reconciled with the Fleetshard sync irrespective of the status (wether deleted or not) of the Kafka.

Reply:
ok. Thanks for explanation.